modules/1_prepare/prepare.tf

```diff
@@ -136,8 +136,8 @@ resource "null_resource" "bastion_init" {
     inline = [
       "sudo chmod 600 ~/.ssh/id_rsa*",
       "sudo sed -i.bak -e 's/^ - set_hostname/# - set_hostname/' -e 's/^ - update_hostname/# - update_hostname/' /etc/cloud/cloud.cfg",
-      "sudo hostnamectl set-hostname --static ${lower(var.cluster_id)}-bastion-${count.index}.${var.cluster_domain}",
-      "echo 'HOSTNAME=${lower(var.cluster_id)}-bastion-${count.index}.${var.cluster_domain}' | sudo tee -a /etc/sysconfig/network > /dev/null",
+      "sudo hostnamectl set-hostname --static ${lower(var.name_prefix)}-bastion-${count.index}.${var.cluster_domain}",
+      "echo 'HOSTNAME=${lower(var.name_prefix)}-bastion-${count.index}.${var.cluster_domain}' | sudo tee -a /etc/sysconfig/network > /dev/null",
```

Comment:
Do we need to use the prefix in hostname as well? I do not think so. let the hostname be with cluster id, just the instance name can have zone.

Reply:
having hostname same as instance name will avoid confusion imho

Reply:
This will need changes in code for helpernode and install playbook vars, also in the external dns as well. Also imo it will add to more confusion with cluster name having the zone. So my suggestion is to stick with zone only in resource naming and not impact the cluster naming.

If we want to use zone everywhere, ie: names, hostnames, cluster name, URIs, then no need to add new variable called name_prefix. Append the zone to cluster_id with no changes elsewhere.

Reply:
Closing this comment as we decided to have zone name in hostname for the nodes.